wiki/App-Initialization-Flow.md

```diff
@@ -0,0 +1,72 @@
+    2. `CommonDomainRoutes.tsx` and `CustomDomainRoutes.tsx` declare the routes to front-end pages, switched in `navigation/Router.tsx` based on results of the `/domain` call.
+    3. "General routes," referring to pages like `/privacy` and `/terms`, are also loaded via `GeneralRoutes.tsx`.
+8. Most of these routes are set to return their relevant views wrapped inside our `LayoutWrapper` component, via the function `withLayout()`, imported from `client/scripts/views/Layout.tsx`. 
+    1. The `LayoutWrapper` injects the page’s specific params into the page as child, and injects the `scope` and `deferChain` arguments into the `Layout` component.
+    2. This `Layout` component includes a nested `Suspense` component (via the `react` library) and `ErrorBoundary` component (via the `react-error-boundary` library).
+    3. The `LayoutComponent` operates as follows:
+        1. If, _at render time_, `app.loadingError` is set, display an application error.
+        2. If, _at render time_, the `app.loginStatusLoaded()` method returns false, a loading spinner is shown, triggered by `shouldShowLoadingState`.
+        3. If, _at render time_, `app.config.chains.getById(scope)` does not return successfully, render a `PageNotFound` view.
+        4. If, _at render time_, there is no `selectedScope`, and no custom domain, then whatever chain is currently being loaded is deinitialized by `deinitChainOrCommunity`, and the `ScopeToLoad` state variable set to `null`. A loading spinner is shown, triggered by `shouldShowLoadingState`.
+        5. If, _at render time_, the `selectedScope` differs from `app.activeChainId()`, the `isLoading` and `scopeToLoad` global state variables are updated (to `true` and `selectedScope`, respectively). `selectChain()` is fired, receiving new scope's chain as argument; on completion, the `isLoading` state variable is set to `false`.
+            - _Note: The `deferChain` param, passed from Route components to the `withLayout()` function, are presently absent from `LayoutAttrs`, where they ought to be passed. If this logic is obsolete, it should be removed from our router files and `layout.tsx` documentation._
+        6. If none of these conditions apply, the routed-to page is rendered.
+        - _Note: A previous iteration of these docs, and of the inline documentation in Layout.tsx, describe conditional logic in which scopes set to Ethereum addresses should trigger a call to `initNewTokenChain`, generating a new community for the provided address while rendering a `LoadingLayout`. The relevant code is no longer present in the file, and this behavior should be considered deprecated unless explicitly reintroduced._
+9. If `selectChain()` (`/helpers/chain.ts`) is fired, per step #8:
+    1. If no `chain` argument is passed, the function defaults to a `chain` value set my `app.user.selectedChain`, or else `app.config.defaultChain`.
+        - _Note: This is likely dead code that can be  removed._
```

Comment:
Technically it's still a valid codepath -- we'd need to inspect all calls to selectChain to confirm that we can remove it.